sstables/storage.cc

```diff
@@ -570,7 +570,7 @@ sstring s3_storage::make_s3_object_name(const sstable& sst, component_type type)
 
     return std::visit(overloaded_functor {
         [&] (const sstring& prefix) -> sstring {
-            return format("/{}/{}", _bucket, sst.filename(type, prefix));
+            return format("/{}/{}/{}", _bucket, prefix, sst.component_basename(type));
```

Comment:
off-topic: if we ever support multi buckets, shouldn't s3_storage::prefix() include bucket as part of the prefix for a component name to distinguish between different buckets? or perhaps if prefix works as a identifier for operations within the storage (meaning we cannot change it without breaking things), maybe we should introduce a location() which returns the full path to a component.

Reply:
prefix() for s3 currently returns table-id if this is keyspace-storage-options thing, or full path including the bucket name if it's an sstable opened by restore code directly from s3. Presumably you're talking about the former case?

Reply:
since we're combining _bucket and _location here for full path, I thought s3_storage::prefix() is not returning bucket since it operates on _location only. I wonder which kind of problems we might have (in the future) when working with multiple buckets, since prefix makes only sense within the bucket it belongs.